src/NAppUpdate.Framework/NAppUpdate.Framework.csproj

```diff
@@ -1,160 +1,16 @@
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\net40\</OutputPath>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <GenerateSerializationAssemblies>Off</GenerateSerializationAssemblies>
-    <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
   </PropertyGroup>
+
   <ItemGroup>
-    <Reference Include="System" />
-    <Reference Include="System.Xml" />
+    <Folder Include="Properties\" />
   </ItemGroup>
+
   <ItemGroup>
-    <Compile Include="Common\INauFieldsHolder.cs" />
-    <Compile Include="Common\Logger.cs" />
-    <Compile Include="Common\NauConfigurations.cs" />
-    <Compile Include="Common\NauFieldAttribute.cs" />
-    <Compile Include="Common\UpdateProcessAsyncResult.cs" />
-    <Compile Include="Common\UpdateProgressInfo.cs" />
-    <Compile Include="Common\WorkScope.cs" />
-    <Compile Include="Conditions\FileChecksumCondition.cs" />
-    <Compile Include="Conditions\FileDateCondition.cs" />
-    <Compile Include="Conditions\FileSizeCondition.cs" />
-    <Compile Include="Conditions\OSCondition.cs" />
-    <Compile Include="Common\Exceptions.cs" />
-    <Compile Include="FeedReaders\AppcastReader.cs" />
-    <Compile Include="Conditions\BooleanCondition.cs" />
-    <Compile Include="Conditions\FileExistsCondition.cs" />
-    <Compile Include="Conditions\FileVersionCondition.cs" />
-    <Compile Include="Conditions\IUpdateCondition.cs" />
-    <Compile Include="Conditions\UpdateConditionAliasAttribute.cs" />
-    <Compile Include="FeedReaders\IUpdateFeedReader.cs" />
-    <Compile Include="FeedReaders\NauXmlFeedReader.cs" />
-    <Compile Include="Resources.Designer.cs">
-      <AutoGen>True</AutoGen>
-      <DesignTime>True</DesignTime>
-      <DependentUpon>Resources.resx</DependentUpon>
-    </Compile>
-    <Compile Include="Sources\IUpdateSource.cs" />
-    <Compile Include="Sources\MemorySource.cs" />
-    <Compile Include="Sources\SimpleWebSource.cs" />
-    <Compile Include="Sources\UncSource.cs" />
-    <Compile Include="Tasks\FileUpdateTask.cs" />
-    <Compile Include="Tasks\IUpdateTask.cs" />
-    <Compile Include="Tasks\RegistryTask.cs" />
-    <Compile Include="Tasks\UpdateTaskAliasAttribute.cs" />
-    <Compile Include="Tasks\UpdateTaskBase.cs" />
-    <Compile Include="UpdateManager.cs" />
-    <Compile Include="Utils\FileChecksum.cs" />
-    <Compile Include="Utils\FileDownloader.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="Utils\NauIpc.cs" />
-    <Compile Include="Utils\FileSystem.cs" />
-    <Compile Include="Utils\PermissionsCheck.cs" />
-    <Compile Include="Utils\ProcessStartFailedException.cs" />
-    <Compile Include="Utils\Reflection.cs" />
+    <PackageReference Include="Microsoft.Windows.Compatibility" Version="2.0.0-preview1-25914-04" />
```

Comment:
If you didn't want this dependency you could consider making the RegistryTask more 'optional'.